### ISSUE TITLE
[Android] Notification Messages can't open from app

### ISSUE THREAD

**Issue by JintoAntony** [invertase/react-native-firebase#3293]
I was trying to implement cloud messaging with react native. I encountered many issues/bugs. 
I did the integration as mentioned in this link : [messaging/quick-start](https://invertase.io/oss/react-native-firebase/v6/messaging/quick-start). After the integration, I can receive messages as push notifications from status bar, only when app is closed.

These are the main issues, I am facing now. 
1.) **Received push notification messages cannot be opened in the application** (App opens when clicking a notification. But, don't know how to get the contents from that message inside the app)
2.)**Can't receive foreground messages** (Code added as per documentation. But, nothing works)

"react": "16.9.0",
"react-native": "0.61.5",
"@react-native-firebase/app": "^6.3.4",
"@react-native-firebase/messaging": "^6.3.4",

**Comment by Ehesp** [comment]
> 1.) Received push notification messages cannot be opened in the application (App opens when clicking a notification. But, don't know how to get the contents from that message inside the app)

This is something which needs to be handled from a notifications package... We're working through the docs at the moment so will be looking into best ways to achieve this.

> 2.)Can't receive foreground messages (Code added as per documentation. But, nothing works)

Can you describe a message? Do you mean a payload or a visible notification? Guessing you've setup the `onMessage` handler?

**Comment by emanu7478** [comment]
On behalf of @JintoAntony we have used the code shown below for foreground messages. But it's not working. when we try to console the **remotMessage.data** it doesn't print anything in the console.
Will you please suggest any solution for this issue ?

```
function App() `{`
  useEffect(() => {
    const unsubscribe = messaging().onMessage(async remoteMessage => {
      console.log('FCM Message Data:', remoteMessage.data);
 
      // Update a users messages list using AsyncStorage
      const currentMessages = await AsyncStorage.getItem('messages');
      const messageArray = JSON.parse(currentMessages);
      messageArray.push(remoteMessage.data);
      await AsyncStorage.setItem('messages', JSON.stringify(messageArray));
    });
  }, []);
}
```

**Comment by Kailash23** [comment]
@emanu7478 How did you send test notification?
```
  initFCMListeners = () => {
    this.unsubscribe = messaging().onMessage(async remoteMessage => {
      console.log('FCM Message Data:', remoteMessage.data);
    });
  };
```
I called this function inside componentDidMount. Not getting any log on test notification (Firebase console -> Grow (Side tab) -> Cloud Messaging -> New Notification

**Comment by Ehesp** [comment]
Ok so currently:

- When sending a Notification (e.g. from console), it'll only show when the app is in the background (Android doing this, not us).
- Notification messages currently do not trigger any event (e.g. `onMessage`).
- There is currently no way of knowing which notification caused the app to open.

Looking through this on the Android side at the moment.

**Comment by Ehesp** [comment]
Just published `6.4.0-rc0`, which includes this functionality for Android. Will get iOS done before going live.

`yarn add @react-native-firebase/app@rc @react-native-firebase/messaging@rc`
